include/TrigScint/QIEInputPulse.h

```diff
@@ -0,0 +1,195 @@
+/**
+ * @file QIEInputPulse.h
+ * @brief Class for the input pulses to QIE
+ * @author Niramay Gogate, Texas Tech University
+ */
+
+#include<vector>
+
+#ifndef TRIGSCINT_QIEINPUTPULSE_H
+#define TRIGSCINT_QIEINPUTPULSE_H
+
+namespace trigscint {
+
+  /**
+   * @class QIEInputPulse
+   * @brief The base class to store the most important functions
+   *
+   * @note Always use the classes inherited from this class for 
+   * QIE input pulse.
+   */
+  class QIEInputPulse
+  {
+ public:
+
+    /**
+     * Evaluate the pulse train at time T
+     */
+    float Eval(float T);
+
+    /**
+     * Evaluate the pulse train at time T
+     */
+    virtual float EvalSingle(float T, int id) = 0;
+
+    /**
+     * Integrate the pulse from T1 to T2
+     */
+    virtual float Integrate(float T1,float T2) = 0;
+
+    /**
+     * Differentiate pulse at time T
+     */
+    virtual float Derivative(float T, int id) = 0;
+
+    /**
+     *  maximum of the pulse
+     */
+    virtual float Max(int id) = 0;
+
+    /**
+     * To add a pulse to the collection
+     * @param toff time at which the pulse starts
+     * @param ampl pulse amplitude (total area under the curve)
+     */
+    void AddPulse(float toff, float ampl);
+
+    /// collection of pulse time offsets
+    std::vector<float> toff_;
+
+    /// collection of pulse amplitudes
+    std::vector<float> ampl_;
+
+    /// no. of pulses in the collection
+    int npulses{0};
+  };
+
+
+  /**
+   * @class Bimoid
+   * @brief Pulse made out of difference of two sigmoids
+   *
+   * @note The pulse maximum is found numerically
+   */
+  class Bimoid : public QIEInputPulse
+  {
+ public:
+    /**
+     * Class constructors.
+     */
+    Bimoid(float start,float qq);
+
+    /**
+     * Default class destructor
+     */
+    ~Bimoid(){};
+    
+    /**
+     * Evaluate the pulse at time T
+     */
+    float EvalSingle(float T, int id) final;
```

Comment:
Since you are overriding the base class, you should specify that here.

Reply:
how should it be specified?

Reply:
Yeah, I have the same question.

Reply:
``` float EvalSingle(float T, int id) final override; ```